plugins/modules/discovery_intent.py

```diff
@@ -200,6 +200,101 @@
                     - Requires valid SSH credentials to work.
                     - Avoid standard ports like 22, 80, and 8080.
                 type: str
+      global_credentials:
+        description:
+            - Credentials that are already created by the user under Device Credentials in Cisco Catalyst Center.
+            - If user doesn't pass any global credentials in the playbook.
+            - By default we will use all the global credentials present in the Cisco Catalyst Center of each type for performing discovery. (Max 5 allowed)
+        type: dict
+        suboptions:
+            cli_credentials_list:
+                description:
+                    - List of Global CLI credentials to be used during device discovery.
+                    - Generally it is advised to create device credentials with unique username or description.
+                type: list
+                elements: dict
+                suboptions:
+                    username:
+                        description: Username for CLI authentication, mandatory when using global CLI credentials.
+                        type: str
+                    description:
+                        description: Name of the CLI credential, mandatory when using global CLI credentials.
+                        type: str
+            http_read_credential_list:
+                description:
+                    - List of Global HTTP Read credentials to be used during device discovery.
```

Comment:
`List of global HTTP Read credentials that will be used in the process of discovering devices.`